src/envoy/auth/jwt_test.cc

```diff
@@ -472,20 +460,28 @@ TEST_F(JwtTestPem, InvalidAlg) {
          Status::ALG_NOT_IMPLEMENTED, nullptr);
 }
 
-TEST_F(JwtTestPem, NonEmptyJwtSubEqual) {
-  DoJwtSubEqualTest(ds.kJwt, ds.kJwtSub, true);
+TEST(JwtSubExtractionTest, NonEmptyJwtSubShouldEqual) {
+  DatasetPem ds;
+  Jwt jwt(ds.kJwt);
+  EXPECT_EQ(jwt.Sub(), ds.kJwtSub);
 }
 
-TEST_F(JwtTestPem, NonEmptyJwtSubNotEqual) {
-  DoJwtSubEqualTest(ds.kJwt, "", false);
+TEST(JwtSubExtractionTest, NonEmptyJwtSubShouldNotEqual) {
+  DatasetPem ds;
+  Jwt jwt(ds.kJwt);
+  EXPECT_NE(jwt.Sub(), "");
 }
 
-TEST_F(JwtTestPem, EmptyJwtSubEqual) {
-  DoJwtSubEqualTest("", "", true);
+TEST(JwtSubExtractionTest, EmptyJwtSubShouldEqual) {
+  DatasetPem ds;
+  Jwt jwt("");
+  EXPECT_EQ(jwt.Sub(), "");
 }
 
-TEST_F(JwtTestPem, EmptyJwtSubNotEqual) {
-  DoJwtSubEqualTest("", ds.kJwtSub, false);
+TEST(JwtSubExtractionTest, EmptyJwtSubShouldNotEqual) {
+  DatasetPem ds;
+  Jwt jwt("");
+  EXPECT_NE(jwt.Sub(), ds.kJwtSub);
```

Comment:
These two tests are redundant too.  A == 0 of course A != 10
Beside "ds" is not used.